feature-libs/commerce-quotes/core/facade/commerce-quotes.service.ts

```diff
@@ -39,6 +53,95 @@ export class CommerceQuotesService implements CommerceQuotesFacade {
       { reloadOn: [CommerceQuotesListReloadQueryEvent] }
     );
 
+  protected createQuoteCommand: Command<
+    { quoteMetadata: QuoteMetadata; quoteComment: Comment },
+    Quote
+  > = this.commandService.create<
+    { quoteMetadata: QuoteMetadata; quoteComment: Comment },
+    Quote
+  >(
+    (payload) =>
+      combineLatest([
+        this.userIdService.takeUserId(),
+        this.activeCartFacade.takeActiveCartId(),
+      ]).pipe(
+        take(1),
+        switchMap(([userId, cartId]) =>
+          zip(
+            of(userId),
+            this.commerceQuotesConnector.createQuote(userId, { cartId })
+          )
+        ),
+        concatMap(([userId, quote]) => {
+          return zip(
+            combineLatest([
+              this.commerceQuotesConnector.editQuote(
+                userId,
+                quote.code,
+                payload.quoteMetadata
+              ),
+              this.commerceQuotesConnector.addComment(
+                userId,
+                quote.code,
+                payload.quoteComment
+              ),
+            ]),
+            of(quote)
+          );
+        }),
+        map(([_, quote]) => quote),
+        tap(() => this.activeCartFacade.reloadActiveCart())
```

Comment:
Based on response in browser I'm not sure if it works correctly. After creating a quote, previous state of cart is still visible after reload. Maybe cart should be reset?
In addition, maybe it's worth to redirect user to page with quote details after creation (this part of feature is already on epic branch)

Reply:
As for now, I find it to be the only viable solution, because we're forced to refresh cart to get new `cartId` due to how quote API works (causing to delete and create exactly the same cart with different id).